lib/iris/coords.py

```diff
@@ -1933,7 +1935,9 @@ def cell(self, index):
 
         return Cell(point, bound)
 
-    def collapsed(self, dims_to_collapse=None):
+    def collapsed(
+        self, dims_to_collapse: Union[int, Sequence[int], None] = None
```

Comment:
@rcomer For this you could use...

```python
from typing import Iterable, Union

# type aliases
Dims = Union[int, Iterable[int]]

def collapsed(
    self, dims_to_collapse: Optional[Dims] = None
```
Which is slightly neater. See https://docs.python.org/3/library/typing.html#typing.Optional

IMHO with typing there certainly seems to be several ways to skin the one cat...